### PR TITLE
Assembly rules

### PR DESCRIPTION
**Pull request by tanaes**
Added rules for assembly via Megahit and Metaspades. 

The choice of assembler(s) is made by a list in `config.yaml`:

```{yaml}
assemblers:
  - metaspades
  - megahit
 ```

All assemblies are executed per sample, then the per-assembler outputs summarized via Quast, then the per-sample Quast outputs summarized by MultiQC.

### CODE REVIEW

File: resources/snakefiles/assemble.smk [hunk 1]
```diff
@@ -35,4 +38,80 @@ rule metaspades_assembly:
 
         """
 
-        
\ No newline at end of file
+rule megahit_assembly:
+    """
+
+    Performs a metagenomic assembly on a sample using MetaSPAdes.
```
Comment (DanielSprockett) [inline resources/snakefiles/assemble.smk:44]: change to say megahit instead of MetaSPAdes

File: resources/config/config.yaml [hunk 2]
```diff
@@ -8,6 +8,10 @@ reads:
   - R1
   - R2
 
+assemblers:
+  - metaspades
```
Comment (DanielSprockett) [inline resources/config/config.yaml:12]: Is this still taking a ridiculous amount of time (i.e. several days) to run?

**Review by DanielSprockett — COMMENTED** [review]
So this looks good, but are we sure we want to use both assembly methods on every sample? At which point do we decide to go with one or the other? Or are we going to bin (as we've previously discussed) using both, and then make a final decision based on quality of the final bins?

**Comment by tanaes** [comment]
> So this looks good, but are we sure we want to use both assembly methods on every sample? At which point do we decide to go with one or the other? Or are we going to bin (as we've previously discussed) using both, and then make a final decision based on quality of the final bins?

Nope, you just choose which one (or both) you want to use in the `config.yaml`. Above I have:

```{yaml}
assemblers:
  - metaspades
  - megahit
```

But if you only wanted to run Megahit:

```{yaml}
assemblers:
  - megahit
```

**Comment by DanielSprockett** [comment]
oh my gosh -- of course it works that way. Sorry, just a bit sleep deprived lately. I will merge this.

**Pull request merged**